scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -1105,7 +1105,7 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
       import scala.concurrent.ExecutionContext.Implicits.global
       this
         .map(bqt.toTableRow)
-        .saveAsBigQuery(
+        .withName(s"${this.tfName.replaceFirst("@.*", "")}$$Write").saveAsBigQuery(
```

Comment:
While this works, a job that has more than one `.saveAsTypedBigQuery` step would be a bit problematic because they would both try to use the same transform name. IIRC when this happens a number is appended onto one of the transform names so that they are unique, but this isn't deterministic and so could cause problems with streaming updates.

I think to get around this, it would be good to incorporate the name set before calling `.saveAsTypedBigQuery`, which right now is used only in the `.map` above. Since the name provider is reset back to the callsite scheme after the map, `this.tfName` as used here isn't useful. So instead, before calling `this.map(bqt.toTableRow)` we could have a `val tfName = this.tfName`, and use that on this line. WDYT?

Reply:
That's a good point, pushed a new commit.